src/Elmish.WPF/ViewModel.fs

```diff
@@ -368,7 +369,13 @@ and [<AllowNullLiteral>] internal ViewModel<'model, 'msg>
             | true, w ->
                 log.LogTrace("[{BindingNameChain}] Closing window", winPropChain)
                 b.WinRef.SetTarget null
-                w.Dispatcher.Invoke(fun () -> w.Close ())
+                (*
+                 * The Window might be in the process of closing,
+                 * so instead of immediately exeucting Window.Close via Dispatcher.Invoke,
+                 * queue a call to Window.Close via Dispatcher.BeginInvoke.
+                 * https://github.com/elmish/Elmish.WPF/issues/330
+                 *)
+                w.Dispatcher.BeginInvoke(DispatcherPriority.Normal, Action(w.Close)) |> ignore
```

Comment:
If you want, I think you can remove the parentheses around the `Action` argument. I usually remove parentheses if there's only a single argument, but do what you want.

```suggestion
                w.Dispatcher.BeginInvoke(DispatcherPriority.Normal, Action w.Close) |> ignore
```

Reply:
No longer relevant